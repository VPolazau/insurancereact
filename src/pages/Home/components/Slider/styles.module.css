.slider {
  width: 100%;
  overflow: hidden;
  position: relative;
}

.sliderWrapper {
  display: flex;
  transition: transform 0.6s ease-in-out;
}

.slide {
  display: flex;
  width: 100%;
  background: linear-gradient(85.86deg, #A9FFDB -4.93%, rgba(126, 155, 255, 0.27) 115.27%);
  height: 500px;
}

.slide:nth-child(even) {  
  background: linear-gradient(85.86deg, rgba(126, 155, 255, 0.27) -6.35%, #A9FFDB 115.27%);
}

.slideContainer {
  margin: 132px auto;  
  display: flex;
  padding: 0 24px;
  /* ширина по макету + 2 падинга */
  max-width: calc(1218px + 48px);
  width: 100%;
  position: relative;
}

.slideContent {
  flex: 0 0 56%;
  width: 100%;
  z-index: 1;
}

.slideImage {
  position: absolute;
  right: 109px;
  bottom: -132px;
  pointer-events: none;
  z-index: 0;
}

/* Стили для картинки второго слайда */
.slide:nth-child(2) .slideImage {
  right: 24px;
}

.slideTitle {
  margin-bottom: 22px;  
  pointer-events: none;
  user-select: none;
}

.slideSubtitle {
  margin-bottom: 40px;  
  pointer-events: none;
  user-select: none;
}

.pagination {
  position: absolute;
  margin-left: auto;
  margin-right: auto;
  left: 0;
  right: 0;
  bottom: 20px;
  display: flex;
  justify-content: center;
  width: max-content;
  gap: 20px;
}

.paginationDot {
  width: 14px;
  height: 14px;
  border-radius: 50%;
  cursor: pointer;
  background-color: transparent;   
  border: 1px solid #22C062;
}

.active {
  background-color: #22C062;
}

@media (max-width: 1150px) {
  .slide {
    height: 376px;
  }

  .slideContainer {
    margin: 64px auto;
  }

  .slideTitle {
    margin-bottom: 15px;  
    pointer-events: none;
    user-select: none;
  }

  .slideSubtitle {
    min-height: 52px;
    margin-bottom: 21px; 
  }

  .slideImage {
    right: 66px;
    bottom: -71px;
  }

  .slideImage > img {
    max-width: 85%;
  }
  
  .slide:nth-child(2) .slideImage {
    right: -101px;
    bottom: -28px;
  }
  
  .slide:nth-child(2) .slideImage > img {
    max-width: 78%;
  }
}

@media (max-width: 965px)  {
  .slideContent {
    min-width: 454px;
  }

  .slideSubtitle {
    margin-bottom: 37px;
  }

  .slideImage {
    right: -48px;
    bottom: -63px;
  }

  .slideImage > img {
    max-width: 77%;
  }
  
  .slide:nth-child(2) .slideImage {
    right: -215px;
    bottom: -1px;
  }
  
  .slide:nth-child(2) .slideImage > img {
    max-width: 59%;
  }
}

@media (max-width: 710px) {
  .slide {
    height: 660px;
    gap: 18px;
  }

  .slideContainer {
    display: flex;
    flex-direction: column;
    justify-content: start;
    margin: 0;
    width: 100%;
    padding: 0 16px;
  }

  .slideContent {
    flex-basis: 40%;
    width: 100%;
    min-width: 0;
    height: 202px;
  }

  .slideTitle {
    font-size: 32px;
    line-height: 130%;
    margin-top: 40px;
    margin-bottom: 16px;
  }

  .slideSubtitle {
    font-size: 20px;
    line-height: 130%;
    margin-top: 16px;
    margin-bottom: 32px;
    max-width: 100%;
  }

  .slideImage {
    flex-basis: 60%;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 0;
  }

  .slide:nth-child(1) .slideImage {
    justify-content: start;
    align-items: center;
    padding: 0 0 7% 0;
    left: 27%;
    width: 76%;
  }

  .slide:nth-child(2) .slideImage {
    justify-content: start;
    align-items: center;
    padding: 0 0 10% 0;
    left: 50px;
    width: 134%;
  }

  .pagination {
    gap: 8px;
  }

  .paginationDot{
    width: 10px;
    height: 10px;
  }
}

@media (max-width: 570px) {
  .slide:nth-child(1) .slideImage {
    justify-content: start;
    align-items: center;
    padding: 0 0 10% 0;
    left: 23%;
    width: 76%;
  }

  .slide:nth-child(2) .slideImage {
    padding: 0 0 15% 0;
    left: 17px;
    width: 154%;
  }
}

@media (max-width: 360px) {
  .slideContent {
    height: 214px;
  }

  .slideTitle {
    font-size: 26px;
    margin-top: 32px;
    margin-bottom: 16px;
  }

  .slideSubtitle {
    font-size: 17px;
  }

  .slide {
    height: 562px;
    gap: 10px;
  }

  .buttonDesign {
    width: 100%;
    justify-content: center;
  }
}
