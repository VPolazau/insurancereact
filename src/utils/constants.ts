import boy from '../assets/img/boy.svg';
import car from '../assets/img/car.svg';

import { INotificationMessage } from '../redux/slices/user/types';

export interface Slides {
    title: string;
    subtitle: string;
    image?: string;
}

export const SLIDES: Slides[] = [
    {
        title: 'Защитите свою квартиру от&nbsp;неожиданных ситуаций',
        subtitle: 'получите страховку не&nbsp;выходя из&nbsp;дома',
        image: boy,
    },
    {
        title: 'Защитите свой автомобиль с&nbsp;нашей страховкой',
        subtitle: 'надежное покрытие от&nbsp;непредвиденных&nbsp;рисков!',
        image: car,
    },
];

export const CARDS = [
    {
        title: 'Широкий спектр услуг',
        description: 'Мы&nbsp;заботимся о&nbsp;Вашей безопасности',
    },
    {
        title: 'Чат поддержки',
        description: 'Получайте быстро и&nbsp;удобно ответы на&nbsp;вопросы',
    },
    {
        title: 'Личный кабинет',
        description: 'Управляйте своей&nbsp;страховкой легко и&nbsp;удобно',
    },
    {
        title: 'Выгодное предложение',
        description: 'Страховая сумма больше, а&nbsp;взнос&nbsp;меньше',
    },
];

export const PLACEHOLDERS = {
    policies: {
        title: 'У&nbsp;Вас пока нет заявок на&nbsp;страхование',
        subTitle:
            'В&nbsp;данном разделе&nbsp;Вы можете найти всю важную информацию, связанную с&nbsp;Вашими заявками на&nbsp;страхование и&nbsp;их&nbsp;статусами',
        buttonText: 'Оформить',
    },
    requests: {
        title: 'У&nbsp;Вас пока нет активных страховых полисов',
        subTitle:
            'Оформите страховой полис, и&nbsp;получите быстрый и&nbsp;удобный доступ ко&nbsp;всей информации о&nbsp;своих полисах в&nbsp;любое удобное для Вас время',
        buttonText: 'Оформить',
    },
    notifications: {
        title: 'У&nbsp;Вас пока нет уведомлений',
        subTitle:
            'В&nbsp;данном разделе&nbsp;Вы можете найти всю важную информацию, связанную с&nbsp;Вашими страховыми полисами, изменениями в&nbsp;Вашем профиле и&nbsp;новостями компании',
        buttonText: 'На главную',
    },
};

// TODO поменять все /in-progress на реальные страницы
export const NAVIGATION_DROP_LINKS = [
    {
        mainLinkPath: '/in-progress',
        mainLinkValue: 'Авто',
        dropNavLinks: [
            { dropLinkPath: '/in-progress', dropLinkValue: 'КАСКО' },
            { dropLinkPath: '/in-progress', dropLinkValue: 'ОСАГО' },
        ],
    },
    {
        mainLinkPath: '/in-progress',
        mainLinkValue: 'Имущество',
        dropNavLinks: [
            { dropLinkPath: '/property/apartment', dropLinkValue: 'Квартира' },
            { dropLinkPath: '/in-progress', dropLinkValue: 'Дом' },
        ],
    },
    {
        mainLinkPath: '/in-progress',
        mainLinkValue: 'Здоровье',
        dropNavLinks: [
            { dropLinkPath: '/in-progress', dropLinkValue: 'ДМС' },
            { dropLinkPath: '/in-progress', dropLinkValue: 'Страхование жизни' },
        ],
    },
    {
        mainLinkPath: '/in-progress',
        mainLinkValue: 'Отдых',
        dropNavLinks: [{ dropLinkPath: '/in-progress', dropLinkValue: 'Страхование путешественников' }],
    },
];

// TODO currentUser это будет пользователь полученный с бэка. "Выход" добавить роут на страницу авторизации
export const USER_DROP_LINKS = {
    currentUser: 'Максим Максимов',
    dropNavLinks: [
        { dropLinkPath: '/account', dropLinkValue: 'Личный кабинет' },
        { dropLinkPath: '/account/policies', dropLinkValue: 'Полисы' },
        { dropLinkPath: '/account/profile', dropLinkValue: 'Мой профиль' },
        { dropLinkPath: '/account/settings', dropLinkValue: 'Настройки' },
        { dropLinkPath: '/', dropLinkValue: 'Выход' },
    ],
};

export const PHONE_NUMBER: string = '+7 (495) 177 - 37 - 37';
export const MAIN_OFFICE_ADDRESS: string = 'г. Москва. Одесская улица 10, 5 этаж, офис 13';
export const COPYRIGHT = '&copy;&nbsp;2023 &nbsp;&laquo;My&nbsp;Insurance&raquo;';
export const IN_PROGRESS =
    'Уважаемые посетители, извините за&nbsp;неудобства, но&nbsp;наш сайт находится в&nbsp;активной разработке.';
export const EMAIL_CONFIRMED = 'Ваш email подтвержден!';

// TODO имитация карточек страховых полисов
export const INSURANCE_CARDS = [
    {
        id: '1',
        property: 'Квартира',
        type: 'Имущество',
        address: 'г. Москва, ул. Зеленоградская, д.102, к.5, кв 135',
        policyNumber: 'WIOO12357909',
        policyValidity: '01.01.2023-01.01.2024',
        policyCost: '17 000 ₽',
        paymentAmount: '8 500 ₽',
        paymentFrequency: 'Раз в 6 месяцев',
    },
    // Дополнительные объекты карточек полисов
];

// TODO имитация списка страховых заявок
export const REQUESTS = [
    {
        id: 1,
        type: 'Страхование квартиры',
        number: 2536324,
        created: '02.11.2022',
        status: 'approved',
        review: '2 дня',
        response: true,
        extension: true,
    },
    {
        id: 2,
        type: 'Страхование дома',
        number: 2834324,
        created: '02.11.2022',
        status: 'rejected',
        review: '2 дня',
        response: true,
    },
    {
        id: 3,
        type: 'КАСКО',
        number: 3534324,
        created: '02.11.2022',
        status: 'pending',
        review: '2 дня',
        response: false,
    },
];

// Имитация уведомлений в модалке
export const NOTIFICATIONS: INotificationMessage[] = [
    {
        id: '1',
        type: 'Страхование квартиры',
        title: 'Ваша заявка на «Страхование квартиры» одобрена',
        date: '04.11.2022',
        isRead: false,
        message:
            'Мы рады сообщить Вам, что Ваша заявка на страхование квартиры была рассмотрена и одобрена. Благодарим за проявленный интерес к нашей компании и надеемся на долгосрочное взаимодействие.',
    },
    {
        id: '2',
        type: 'Страхование дома',
        title: 'Ваша заявка на «Страхование дома» отклонена',
        date: '05.09.2022',
        isRead: false,
        message:
            'Уважаемый Константин Александрович! К сожалению, мы не можем удовлетворить вашу заявку на страхование дома...',
    },
    {
        id: '3',
        type: 'Смена пароля',
        title: 'Ваш пароль успешно изменен',
        date: '05.09.2022',
        isRead: true,
        message: 'Здравствуйте! Мы рады сообщить Вам, что пароль для вашей учетной записи был успешно изменен',
    },
];

export const RadioBtns = [
    { value: 'on', title: 'Вкл' },
    { value: 'off', title: 'Выкл' },
];

export const CONFIRM_POPUP_TITLES = {
    request: 'Вы действительно хотите отозвать&nbsp;заявку?',
    notification: 'Вы действительно хотите удалить&nbsp;уведомление?',
};

// TODO контент модалки ответа
export const RESPONSE_POPUP_CONTENT = {
    commonText: {
        title: 'Добрый день!',
        signature: 'С&nbsp;уважением, компания MyInsurance',
    },
    approved: {
        message:
            'Мы&nbsp;рады сообщить, что Ваша заявка на&nbsp;&laquo;Страхование&nbsp;квартиры&raquo; была рассмотрена и&nbsp;одобрена. Благодарим за&nbsp;проявленный интерес к&nbsp;нашей компании и&nbsp;надеемся на&nbsp;долгосрочное взаимодействие.',
        requisitesTitle: 'Реквизиты для оплаты:',
        requisites: [
            {
                label: 'Название компании',
                value: 'СПАО «My Insurance»',
            },
            {
                label: 'Расчетный счёт',
                value: '40704917073289876529',
            },
            {
                label: 'БИК',
                value: '011111111',
            },
            {
                label: 'ИНН',
                value: '7711111111',
            },
        ],
    },
    rejected: {
        message:
            'Уважаемый Константин Александрович! К&nbsp;сожалению, мы&nbsp;вынуждены сообщить Вам об&nbsp;отказе на&nbsp;заявку &laquo;Страхование дома&raquo;',
        additionalMessage:
            'Если у&nbsp;Вас возникнут дополнительные вопросы или&nbsp;потребности в&nbsp;страховании, пожалуйста, свяжитесь с&nbsp;нами, и&nbsp;мы&nbsp;будем рады помочь Вам.',
    },
};

export const INSURANCE_CASE = [
    {
        id: '1',
        type: 'Общая информация',
    },
    {
        id: '2',
        type: 'Имущество',
    },
    {
        id: '3',
        type: 'Здоровье',
    },
    {
        id: '4',
        type: 'Путешествия',
    },
    {
        id: '5',
        type: 'Автострахование',
    },
];
