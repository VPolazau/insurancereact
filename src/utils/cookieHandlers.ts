export const setCookie = (name: string, value: string) => {
    const updatedCookie = `${name}=${value}`;
    document.cookie = `${updatedCookie}`;
};

export const getCookie = (name: string) => {
    const matches = document.cookie.match(
        new RegExp(`(?:^|; )${name.replace(/([.$?*|{}()[\]\\/+^])/g, '\\$1')}=([^;]*)`)
    );
    return matches ? matches[1] : undefined;
};
