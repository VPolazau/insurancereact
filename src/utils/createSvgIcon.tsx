import React from 'react';

import { SvgIcon } from '../components/SvgIcon';

export interface ISvgIconProps extends React.SVGProps<SVGSVGElement> {
    titleAccess?: string;
}
export const createSvgIcon = (content: React.ReactNode, viewBox?: string): React.FC<ISvgIconProps> => {
    return function (props) {
        return (
            <SvgIcon viewBox={viewBox} {...props}>
                {content}
            </SvgIcon>
        );
    };
};
