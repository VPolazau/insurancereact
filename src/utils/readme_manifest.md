Должны быть такие разрешения для PWA icons 512/152/144/120/114/76/72/60/57