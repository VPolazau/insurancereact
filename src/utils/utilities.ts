export function decodeHtmlEntities(input: string): string {
    const parser = new DOMParser();
    const decodedString = parser.parseFromString(`<!DOCTYPE html><body>${input}`, 'text/html').body.textContent;
    return decodedString || '';
}
