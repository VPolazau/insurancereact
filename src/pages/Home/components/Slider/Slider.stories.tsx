import { Story, Meta } from '@storybook/react';

import { SLIDES } from '../../../../utils/constants';

import { Slider, SliderProps } from './Slider';

export default {
    title: 'UI/Slider',
    component: Slider,
} as Meta;

const Template: Story<SliderProps> = (args) => <Slider {...args} />;

export const Default = Template.bind({});
Default.args = {
    slides: SLIDES,
};
