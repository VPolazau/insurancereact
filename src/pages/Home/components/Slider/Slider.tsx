import React, { useState, useRef, MouseEvent, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import cn from 'classnames';

import { Typography, Button } from '../../../../components';
import { Slides } from '../../../../utils/constants';
import { decodeHtmlEntities } from '../../../../utils/utilities';

import styles from './styles.module.css';

export interface SliderProps {
    slides: Slides[];
}

export const Slider: React.FC<SliderProps> = ({ slides }) => {
    const [activeSlide, setActiveSlide] = useState<number>(0);
    const [isDragging, setIsDragging] = useState<boolean>(false);
    const [dragStartX, setDragStartX] = useState<number>(0);
    const [dragDistance, setDragDistance] = useState<number>(0);
    const sliderRef = useRef<HTMLDivElement>(null);
    const slideWidth = 100 / slides.length;
    const dragThreshold = 100; // Пороговое значение для определения переключения слайда
    const navigate = useNavigate();

    const handlePaginationClick = (index: number) => {
        setActiveSlide(index);
    };

    const handleMouseDown = (e: MouseEvent<HTMLDivElement>) => {
        setIsDragging(true);
        setDragStartX(e.clientX);
    };

    const handleMouseMove = (e: MouseEvent<HTMLDivElement>) => {
        if (!isDragging) return;
        const slider = sliderRef.current;
        if (slider) {
            const newDragDistance = e.clientX - dragStartX;
            setDragDistance(newDragDistance);
        }
    };

    const handleMouseUp = () => {
        setIsDragging(false);

        if (Math.abs(dragDistance) > dragThreshold) {
            if (dragDistance > 0 && activeSlide > 0) {
                setActiveSlide(activeSlide - 1);
            } else if (dragDistance < 0 && activeSlide < slides.length - 1) {
                setActiveSlide(activeSlide + 1);
            }
        }

        setDragDistance(0);
    };

    const handleMouseLeave = () => {
        setIsDragging(false);
        setDragDistance(0);
    };

    // TODO
    const handleApply = () => {
        navigate('/in-progress');
    };

    useEffect(() => {
        // Автоматическое переключение слайда каждые 10 секунд
        const timer = setTimeout(() => {
            const nextSlide = (activeSlide + 1) % slides.length;
            setActiveSlide(nextSlide);
        }, 10000);

        return () => {
            clearTimeout(timer);
        };
    }, [activeSlide, slides.length]);

    return (
        <div
            data-testid="Slider"
            className={styles.slider}
            onMouseDown={handleMouseDown}
            onMouseMove={handleMouseMove}
            onMouseUp={handleMouseUp}
            onMouseLeave={handleMouseLeave}
            ref={sliderRef}
        >
            <div
                className={styles.sliderWrapper}
                style={{
                    width: `${slides.length * 100}%`,
                    transform: `translateX(calc(-${activeSlide * slideWidth}% + ${dragDistance}px))`,
                }}
            >
                {slides.map((slide, index) => (
                    <div className={styles.slide} key={index}>
                        <div className={styles.slideContainer}>
                            <div className={styles.slideContent}>
                                <Typography variant="h1" className={styles.slideTitle} data-testid="sliderTitle">
                                    {decodeHtmlEntities(slide.title)}
                                </Typography>
                                <Typography variant="h3" className={styles.slideSubtitle} data-testid="sliderSubtitle">
                                    {decodeHtmlEntities(slide.subtitle)}
                                </Typography>
                                <Button variant="primary" onClick={handleApply} className={styles.buttonDesign}>
                                    Оформить
                                </Button>
                            </div>
                            <div className={styles.slideImage}>
                                <img src={slide.image} alt="" />
                            </div>
                        </div>
                    </div>
                ))}
            </div>
            <div className={styles.pagination}>
                {slides.map((_, index) => (
                    <button
                        data-testid="sliderButton"
                        key={index}
                        className={cn(styles.paginationDot, { [styles.active]: index === activeSlide })}
                        onClick={() => handlePaginationClick(index)}
                    />
                ))}
            </div>
        </div>
    );
};
