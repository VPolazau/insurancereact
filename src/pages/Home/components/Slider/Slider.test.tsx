import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, fireEvent, cleanup, screen } from '@testing-library/react';

import { decodeHtmlEntities } from '../../../../utils/utilities';

import { SLIDES } from '../../../../utils/constants';

import { Slider } from './Slider';

afterEach(cleanup);

describe('Slider', () => {
    it('slider 1v', () => {
        render(
            <MemoryRouter>
                <Slider slides={SLIDES} />
            </MemoryRouter>
        );

        const titles = screen.getAllByTestId('sliderTitle');
        const subtitles = screen.getAllByTestId('sliderSubtitle');
        const button = screen.getAllByTestId('sliderButton');

        const mokText = {
            title0: decodeHtmlEntities(SLIDES[0].title),
            subtitle0: decodeHtmlEntities(SLIDES[0].subtitle),
            title1: decodeHtmlEntities(SLIDES[1].title),
            subtitle1: decodeHtmlEntities(SLIDES[1].subtitle),
        };

        expect(screen.getByTestId('Slider')).toBeInTheDocument();
        expect(titles[0].textContent).toBe(mokText.title0);
        expect(subtitles[0].textContent).toBe(mokText.subtitle0);

        expect(titles[1].textContent).toBe(mokText.title1);
        expect(subtitles[1].textContent).toBe(mokText.subtitle1);

        expect(button[0].classList.contains('active')).toBe(true);
        expect(button[1].classList.contains('active')).toBe(false);
        fireEvent.click(button[1]);
        expect(button[0].classList.contains('active')).toBe(false);
        expect(button[1].classList.contains('active')).toBe(true);
        fireEvent.click(button[0]);
        expect(button[0].classList.contains('active')).toBe(true);
        expect(button[1].classList.contains('active')).toBe(false);

        expect(screen.getByTestId('Slider')).toMatchSnapshot();
    });
});
