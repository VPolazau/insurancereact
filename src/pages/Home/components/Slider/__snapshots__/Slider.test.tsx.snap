// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider slider 1v 1`] = `
<div
  class="slider"
  data-testid="Slider"
>
  <div
    class="sliderWrapper"
    style="width: 200%; transform: translateX(calc(-0% + 0px));"
  >
    <div
      class="slide"
    >
      <div
        class="slideContainer"
      >
        <div
          class="slideContent"
        >
          <div
            class="text h1 slideTitle"
            data-testid="sliderTitle"
          >
            Защитите свою квартиру от неожиданных ситуаций
          </div>
          <div
            class="text h3 slideSubtitle"
            data-testid="sliderSubtitle"
          >
            получите страховку не выходя из дома
          </div>
          <button
            class="buttonDesign btn primary"
            data-testid="button"
          >
            Оформить
          </button>
        </div>
        <div
          class="slideImage"
        >
          <img
            alt=""
            src="boy.svg"
          />
        </div>
      </div>
    </div>
    <div
      class="slide"
    >
      <div
        class="slideContainer"
      >
        <div
          class="slideContent"
        >
          <div
            class="text h1 slideTitle"
            data-testid="sliderTitle"
          >
            Защитите свой автомобиль с нашей страховкой
          </div>
          <div
            class="text h3 slideSubtitle"
            data-testid="sliderSubtitle"
          >
            надежное покрытие от непредвиденных рисков!
          </div>
          <button
            class="buttonDesign btn primary"
            data-testid="button"
          >
            Оформить
          </button>
        </div>
        <div
          class="slideImage"
        >
          <img
            alt=""
            src="car.svg"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="pagination"
  >
    <button
      class="paginationDot active"
      data-testid="sliderButton"
    />
    <button
      class="paginationDot"
      data-testid="sliderButton"
    />
  </div>
</div>
`;
